### PR TITLE
Need to check that response has not been already sent in phalcon/mvc/micro.zep

### PR DESCRIPTION
**Pull request by Fabsolute**
Hello!

* Type: bug fix

**In raising this pull request, I confirm the following :**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

It is not possible to return a string value from a mapped function for an HTTP Method, and then catch that returned value by using Micro::getReturnedValue() method inside handler on Micro::after($handler), its overriding my response.

Thanks

### CODE REVIEW

**Comment by Jurigag** [comment]
This is bug fix. It should be made to 3.0.x, also update changelog.

**Comment by Fabsolute** [comment]
Changelog updated.

**Comment by Jurigag** [comment]
I mean unit test.

**Review by sergeyklay — CHANGES_REQUESTED** [review]
@Fabsolute In general I like this PR. Could you please provide a small test which covers these changes?

**Review by sergeyklay — CHANGES_REQUESTED** [review]
@Fabsolute 

```
FILE: /home/travis/build/phalcon/cphalcon/tests/unit/Mvc/MicroTest.php
----------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 451 | ERROR | [x] Expected 1 space after USE keyword; found 0
     |       |     (Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterUse)
 451 | ERROR | [x] Expected 1 space before opening brace; found 0
     |       |     (Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeBrace)
 458 | ERROR | [x] Expected 1 space before opening brace; found 0
     |       |     (Squiz.Functions.MultiLineFunctionDeclaration.SpaceBeforeBrace)
 459 | ERROR | [x] Line indented incorrectly; expected at least 24
     |       |     spaces, found 23
     |       |     (Generic.WhiteSpace.ScopeIndent.Incorrect)
 467 | ERROR | [x] The closing brace for the class must go on the next
     |       |     line after the body
     |       |     (PSR2.Classes.ClassDeclaration.CloseBraceAfterBody)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 5 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

File: tests/unit/Mvc/MicroTest.php [hunk 1]
```diff
@@ -440,4 +440,27 @@ function () {
...
+                        $app->response->setJsonContent($content)->send();
+                    }
+                );
+                $app->map(
+                    "/api",
+                    function () {
+                        return "success";
+                    }
+                );
+                $app->handle("/api");
```
Comment (sergeyklay) [inline tests/unit/Mvc/MicroTest.php]:
```php
expect($micro->handle('/api'))->equals("success"); //❓ 
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
Fixed in the `3.2.x` branch. Thank you for contributing.

**Pull request closed without merging**